babel/util.py

```diff
@@ -205,10 +206,29 @@ class TextWrapper(textwrap.TextWrapper):
         r'(?<=[\w\!\"\'\&\.\,\?])-{2,}(?=\w))',   # em-dash
     )
 
+    # e.g. '\u2068foo bar.py\u2069:42'
+    _enclosed_filename_re = re.compile(r'(\u2068[^\u2068]+?\u2069(?::-?\d+)?)')
+
+    def _split(self, text):
+        """Splits the text into indivisible chunks while ensuring that file names
+        containing spaces are not broken up.
+        """
+        enclosed_filename_start = '\u2068'
+        if enclosed_filename_start not in text:
+            # There are no file names which contain spaces, fallback to the default implementation
+            return super()._split(text)
+
+        super_ = super() #  Python 3.9 fix, super() does not work in list comprehension
+        chunks = re.split(self._enclosed_filename_re, text)
+        chunks = [[c] if c.startswith(enclosed_filename_start) else super_._split(c) for c in chunks]
+        chunks = [c for c in chain.from_iterable(chunks) if c]
+        return chunks
```

Comment:
This is kind of hard to read...

Something like (nb: didn't test this)

```suggestion
        chunks = []
        for chunk in re.split(self._enclosed_filename_re, text):
            if chunk.startswith(enclosed_filename_start):
                chunks.append(chunk)
            else:
                chunks.extend(super()._split(chunk))
        return [c for c in chunks if c]
```

smells more [Zen of Python](https://peps.python.org/pep-0020/) to me and I have the feeling it could also be more performant.

Reply:
yep that looks more digestible 😅